server/server.go

```diff
@@ -82,6 +91,16 @@ func New(cfg Config) (*Server, error) {
 	}
 
 	httpClient := &http.Client{}
+	var (
+		reloadNotifier *reloadNotifier
```

Comment:
either export the `reloadNotifier` type or name the variable different since you are shadowing the type with this variable